frontend/src/Components/EventCard.jsx

```diff
@@ -185,10 +185,13 @@ const EventCard = (props) => {
                                     <span>Request pending</span>
                                     <button className='btn btn-primary float-right' onClick={deleteVolunteerForEvent} disabled={waitingForRender}>Remove</button>
                                 </div>
-                            : loggedUser && loggedUser.v_id
-                                ? <div className='card-text text-right'>
-                                    <button className='btn btn-primary' onClick={volunteerForEvent} disabled={waitingForRender}>Volunteer for this event</button>
-                                </div>
+                            : loggedUser && loggedUser.v_id && Date.now() < event.event_end
```

Comment:
Date.now() will give a number, you're comparing a number to a string
you can do `Date.now() < new Date(event.event_end).getTime()`